utils/mariadb/files/mysqld.init

```diff
@@ -36,18 +39,17 @@ start_service() {
 		return 1
 	}
 
-	mkdir -p /var/lib/mysql
-	chown "$my_user":"$my_group" /var/lib/mysql
-
-	mkdir -p /var/run/mysqld
-	chown "$my_user":"$my_group" /var/run/mysqld
+	for i in /var/lib/mysql "$datadir" "$tmpdir" "$(dirname "$pidfile")" "$(dirname "$socketfile")"; do
+		mkdir -p "$i"
+		chown "$my_user":"$my_group" "$i"
+	done
 
 	procd_open_instance
 
 	procd_set_param command $PROG $MY_ARGS
-	procd_set_param pidfile /var/run/mysqld.pid
 	# forward stderr to logd
 	procd_set_param stderr 1
+	procd_set_param user "$my_user"
```

Comment:
No, this is not common. It doesn't drop privileges and when you take a look at [upstream recommended](https://github.com/MariaDB/server/blob/10.4/support-files/mariadb.service.in#L43) way of starting it, they start it as user. Also when you read [MySQL documentation](https://dev.mysql.com/doc/refman/5.7/en/changing-mysql-user.html) it clearly states 

>   However, you should avoid running the server as the Unix root user for security reasons

Starting and running it as root poses security issue as you can then potentially as mysql root change anything on filesystem.

Reply:
But it does. This is what happens after doing a server install on a fresh system:

mysql_install_db --force --basedir=/usr
/etc/init.d/mysqld  start
ps w|grep mysqld
 3010 mariadb   330m S    /usr/bin/mysqld

This is with defaults file in /etc/default/mysqld unchanged. The server switched to user mariadb automatically (because that is the user for mysqld in the default my.cnf.

So procd starts the server as root, it's able to read the 0600 my.conf just fine, and then drops privs eventually. So we're not _running_ the server as root by default.

Reply:
Ah, that explains it - I use my own configuration. I still think it is not a good idea to depend on users not deleting by accident something in default configuration and be safe by default especially when there are not many real benefits of the other way.